### ISSUE TITLE
Memory efficient MAML

### ISSUE THREAD

**Issue by dbaranchuk** [cbfinn/maml#72]
Hi,

We would like to share the application of the gradient checkpointing technique to MAML. This allows performing way (~10-100x) more MAML steps with the same GPU memory budget. Maybe someone dealing with MAML will find it useful for their projects.

[PyTorch implementation](https://github.com/dbaranchuk/memory-efficient-maml)

Example:
[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/dbaranchuk/memory-efficient-maml/blob/master/example.ipynb)

Any comments and proposals are highly appreciated.
